doc/dart_documentation_comment_specification.md

```diff
@@ -0,0 +1,478 @@
+# **Dart Documentation Comments Specification**
+Version: 1.0.0
+
+[TOC]
+
+## **1\. Introduction**
+
+### **1.1. Purpose**
+
+The purpose of this document is to provide a single specification for documentation comments in the Dart programming language. Its goal is to ensure that documentation is parsed consistently by all tools and to serve as a definitive standard for developers and tool authors.
+
+### **1.2. Scope**
+
+**In Scope**
+
+This specification covers the following topics:
+
+* The *syntax* for writing documentation comments.
+* The rules governing where documentation comments can be *placed*.
+* The mechanism for *resolving* references between elements.
+* The logic for *associating* documentation with code elements.
+
+**Out of Scope**
+
+This document does not cover:
+
+* The implementation details of specific documentation tools (e.g., `dartdoc`, `analysis server`).
+* The syntax and behavior of tool-specific directives (e.g., `@template`, `@canonicalFor`).
+* Any details related to the visual styling or HTML layout of generated documentation.
+* Best practices or style guides for writing effective documentation content.
+
+### **1.3. Terminology**
+
+* **Doc Comment:** A comment intended to be processed by documentation generation tools, like  dartdoc.
+* **Element**: A specific, declared element in the Dart code, such as a class, method, function, variable, or type parameter.
```

Comment:
Maybe worth it to go into this here, or later, but what about "induced" elements (that is the analyzer term; not sure if it is in the spec)?

So this would be `int x = 1;` inducing a getter and a setter, and `final x = 1;` inducing only a getter. Is one the element? Are both?

If a class A declares `int x = 1;` and then a subclass B declares `set x(int value) {}`, then the property `B.x` is declared half in A and half in B. If I link to `[B.x]`, does that link to the getter indueced by the property `A.x`? Etc. etc.